include/dylib.hpp

```diff
@@ -139,6 +143,17 @@ class dylib {
 
     explicit dylib(const char *name, bool decorations = add_filename_decorations)
         : dylib("", name, decorations) {}
+
+#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
+    explicit dylib(const std::filesystem::path &path)
+        : dylib("", path.string().c_str(), no_filename_decorations) {}
+
+    dylib(const std::filesystem::path &dir_path, const std::string &name, bool decorations = add_filename_decorations)
```

Comment:
Maybe rename `name` -> `base_filename` or `base_name`?

Reply:
Mhhh or `lib_name` ?

Reply:
@martin-olivier : Sure, I guess. But - did I mention you should doxygen the library?

Reply:
You are right, I forgot to do that, I'm gonna open a separate issue 👍

Reply:
I mentioned this because, regardless of the choice, this function could use a doxygen comment explaining what the parameters mean.

Reply:
Mhhh ok, I was thinking you meant generate doxygen documentation from source code (Doxyfile ...)
I updated the name in [f86045f](https://github.com/martin-olivier/dylib/pull/53/commits/f86045fa45eddb5a9c191cbfc47680612bcc5b35)